### ISSUE TITLE
Use PlaySound(e,v) in a significant number of characters alter the player movement. 

### ISSUE THREAD

**Issue by Dimoxinil** [TheGameCreators/GameGuruRepo#299]
I know this might sound strange, but..

About a month ago, I started noticing that the player mobility worsened as we move forward in the game. It's almost impossible to play.
I started runing every test imaginable..  With time I discovered that it was related to the character entities. Every time the player faced a new deal of enemies the player mobility worsened.
I thought it was a error in my code. I reviewed several times my Lua scripts and I didn't find the problem.

This last Friday I found and insolate the problem..

Any time a new character use the Lua commnad PlaySound(e,v).. give an bug in the engine. At first almost imperceptible, but with each new spawed character using PlaySound(e,v).. the player mobility worsened.

I made a video to better understand them. 

[Video link](https://www.youtube.com/watch?v=4H9IFo8GNuk&feature=youtu.be)

1º I turn around a concrete girder.. all ok
2º A bunch of zombies use the Lua commnad PlaySound(e,v).

This is the simple script used in the zombies:

         if g_Entity[e]['activated'] == 1 then
	    SetActivated(e,2)			   
            PlaySound(e,0)
            PlaySound(e,1)
            PlaySound(e,2)
            PlaySound(e,3)
            PlaySound(e,4)
         end

3º	I turn around a concrete girder.. again.

Now you can appreciate  the deterioration of the player mobility.

I attach a zip file with the map and the scripts used.
``
[Dropbox link](https://www.dropbox.com/s/b47vcl6s7ss59rp/Files.zip?dl=0#)

**Is anyone else having this issue?**

**Comment by AmenMoses** [comment]
How long are the sound samples?  If every zombie is playing 5 sound samples I'm not surprised that performance is poor.

Oh and the dropbox link doesn't work for me, although it wouldn't of been much use unless it contains the sound samples as well.

**Comment by Dimoxinil** [comment]
The sounds are the same that the entity (fatZombie) has associated by default.

soundset      = audiobank\zombies\Fat_Male_Zombie_Groan_1.wav     359KB
soundset1     = audiobank\zombies\Fat_Male_Zombie_Attack.wav       130KB
soundset2     = audiobank\zombies\Fat_Male_Zombie_Groan_2.wav    157KB
soundset3     = audiobank\zombies\Fat_Male_Zombie_Death.wav        240KB
soundset4     = audiobank\zombies\zombie_melee_hit.wav                   61KB

My apologies, the links are now corrected.

[Video youtube](https://www.youtube.com/watch?v=4H9IFo8GNuk&feature=youtu.be)

[Dopbox Files.zip ](https://www.dropbox.com/s/b47vcl6s7ss59rp/Files.zip?dl=0#)

**Comment by Dimoxinil** [comment]
The problem affects **only to the 3d sounds**. 

 In other words, the wav(mono) files used by engine that sound in the  entity position  associated with the sound.

**Comment by AmenMoses** [comment]
Weird, doesn't appear to effect the frame rate though!   Or at least the reported frame rate.

**Comment by Dimoxinil** [comment]
The problem getting worse with every new entity that emits 3d sound, to the point of being unplayable..
but the framerate is completely stable.

**Comment by Dimoxinil** [comment]
I'm sorry, the death of a close relative... well .. I don't find any
motivation at this time.

I have stopped working with GameGuru and I don't know when I'll take it back.

Best regards.

**Comment by synchromesh62** [comment]
@LeeBamberTGC 
@plemsoft 
Video and test map supplied above

**Comment by LeeBamberTGC** [comment]
Tried with VSYNC off and on, all manner of tests with the provided FPM. The issue resembles a bug fixed many moons ago relating to 3D sounds hitting performance quite badly due to the update call each cycle. This is likely the same old bug. I have posted a shot with VSYNC off to finish my own tests and can report no stutter, only a smooth movement after following the above instructions:

![image](https://user-images.githubusercontent.com/35840583/77129774-27253f80-6a4d-11ea-9230-dd068f590910.png)
